lms/djangoapps/teams/csv.py

```diff
@@ -321,6 +321,44 @@ def validate_proposed_team_size_wont_exceed_maximum(self, team_name, teamset_id)
         self.user_count_by_team[(teamset_id, team_name)] += 1
         return True
 
+    def remove_user_from_existing_memberships(self, row):
+        """
+        Remove a user from a team if:
+        a. The user's current team is different from the team specified in csv for the same teamset (this user will
+           then be assigned to a new team in 'add_user_to_team`.
+        b. The team value in the CSV is blank - the user should be removed from the current team in teamset.
+        """
+        for ts_id in self.teamset_ids:
+            if row[ts_id] is None:
+                # remove this student from the teamset
+                try:
+                    membership = CourseTeamMembership.objects.get(
+                        user_id=row['user'].id,
+                        team__topic_id=ts_id
+                    )
+                    membership.delete()
+                except CourseTeamMembership.DoesNotExist:
+                    pass
+            else:
+                # reassignment happens only if proposed team membership is different from existing team membership
+                if (row['user'].id, ts_id) in self.existing_course_team_memberships:
+                    current_user_teams_name = self.existing_course_team_memberships[row['user'].id, ts_id].name
```

Comment:
I'm confused by this line. I thought self.existing_course_team_memberships only mapped to team ids?

Reply:
whoops, should have saved my review until i finished reading :p